Cython/Compiler/ExprNodes.py

```diff
@@ -7340,6 +7357,14 @@ def generate_result_code(self, code):
                                         '"Memoryview is not initialized");'
                         '%s'
                     '}' % (self.result(), code.error_goto(self.pos)))
+        elif self.entry.is_cpp_optional and self.initialized_check:
+            null_code = self.type.cpp_optional_check_for_null_code(self.result())
+            code.putln(
+                'if (unlikely(!%s)) {'
+                    'PyErr_SetString(PyExc_AttributeError,'
+                                    '"C++ class attribute is not initialized");'
+                    '%s'
+                '}' % (null_code, code.error_goto(self.pos)))
```

Comment:
This should also go into `code.put_error_if_unbound()` (and include the attribute name).